src/overlays/actors/ovl_En_Tanron3/z_en_tanron3.c

```diff
@@ -28,39 +36,380 @@ const ActorInit En_Tanron3_InitVars = {
-    { ELEMTYPE_UNK3, { 0xF7CFFFFF, 0x00, 0x02 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_ON | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_HIT3,
+        AT_ON | AT_TYPE_ENEMY,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK3,
+        { 0xF7CFFFFF, 0x00, 0x02 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_ON | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
     { 7, 10, -5, { 0, 0, 0 } },
 };
 
-#endif
+// This actor has two colliders (one for AC and one for AT), but uses the same
+// ColliderCylinderInit for both of them, leaving this one totally unused.
+static ColliderCylinderInit sUnusedCylinderInit = {
+    {
+        COLTYPE_HIT3,
+        AT_ON | AT_TYPE_ENEMY,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK3,
+        { 0xF7CFFFFF, 0x00, 0x02 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_ON | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
+    { 20, 20, -10, { 0, 0, 0 } },
+};
+
+extern FlexSkeletonHeader D_0600DA20;
+extern AnimationHeader D_0600DAAC;
+
+void EnTanron3_CreateEffect(GlobalContext* globalCtx, Vec3f* effectPos) {
+    UnkTanron3Effect* effectPtr = (UnkTanron3Effect*)globalCtx->specialEffects;
+    s16 i;
+
+    for (i = 0; i < 150; i++, effectPtr++) {
+        if ((effectPtr->unk_00 == 0) || (effectPtr->unk_00 == 1)) {
+            effectPtr->unk_00 = 2;
+            effectPtr->unk_04 = *effectPos;
+            effectPtr->unk_10 = *sZeroVec;
+            effectPtr->unk_1C = *sZeroVec;
+            effectPtr->unk_1C.y = -2.0f;
+            effectPtr->unk_34.x = 0.1f;
+            effectPtr->unk_34.y = 0.0f;
+            effectPtr->unk_34.z = Rand_ZeroFloat(M_PI * 2);
+            effectPtr->unk_02 = Rand_ZeroFloat(100.0f);
+            effectPtr->unk_10.x = randPlusMinusPoint5Scaled(25.0f);
+            effectPtr->unk_10.z = randPlusMinusPoint5Scaled(25.0f);
+            return;
+        }
+    }
+}
+
+void EnTanron3_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTanron3* this = THIS;
+
+    this->actor.gravity = -1.0f;
+    Collider_InitAndSetCylinder(globalCtx, &this->atCollider, &this->actor, &sCylinderInit);
+    Collider_InitAndSetCylinder(globalCtx, &this->acCollider, &this->actor, &sCylinderInit);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600DA20, &D_0600DAAC, this->jointTable, this->morphTable, 10);
+    Actor_SetScale(&this->actor, 0.02f);
+    EnTanron3_SetupAct(this, globalCtx);
+    this->actor.flags &= ~1;
+    this->currentRotationAngle = Rand_ZeroFloat(500000.0f);
+    this->waterSurfaceYPos = 430.0f;
+    sGyorg = (Boss03*)this->actor.parent;
+}
+
+void EnTanron3_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    sGyorg->unk_252--;
```

Comment:
If I were to guess, this is the number spawned? Could you at least leave comments on the struct if you have an idea what these do?

Reply:
Yeah, it's probably number spawned. That said, it's for a different actor and thus that documentation isn't required.

Would recommend to just update it, though.

Reply:
It's not the number spawned; it's the number currently alive (which is the number spawned *when they first spawn*). If you kill some of the fishes, this number goes down; if you kill *all* of them, it becomes -1. I'll try to give this a good name in the Boss03 struct.

Reply:
Oh yeah, that's what I meant lol, been a long day.